content/projects/gsoc/2021/projects/git-credentials-binding.adoc

```diff
@@ -47,23 +49,34 @@ They will be able to create and delete remote branches.
 They will be able to use the git command line options of their choice, including `--single-branch` and `--recurse-submodules`.
 Users will be able to run authenticated git commands in their Jenkins Pipelines without modifying the git plugin.
 
-=== Implementation examples
+=== Implementation
+Both `gitUsernamePassword` and `gitSshPrivateKey` bindings use the https://plugins.jenkins.io/credentials/[Credential Plugin]
+to retrieve user's credential using the Credentials API, the https://plugins.jenkins.io/credentials-binding/[Credential Binnding Plugin]
+is used to bind Git specific environment variables with programs to perform authentication on behalf of the user, without their interaction with the command-line.
+
+==== Git Username and Password Binding
+
+The `gitUsernamePassword` implementation use the Jenkins username and password
+values retrieved through Credential API, to access a remote repository  over *HTTP* protocol. This binding use the
+`GIT_ASKPASS` environment variable to provide credentials requested by `Git CLI`, program specified by this variable is invoked with a suitable prompt on the command-line, and the user’s credential are read from its standard output.
+
+image::https://gist.githubusercontent.com/arpoch/96c75cc32afd0e50476e273bd0269a86/raw/7875f70561e1417721ec7f637fa6f160a27822a5/UsernameandPassword.png[Git Username and Password]
```

Comment:
I believe that the preference is to include images in the repository rather than reference them as a gist from outside the repository.  Could you place a copy of this image in the correct images directoy?
```suggestion
image::https://gist.githubusercontent.com/arpoch/96c75cc32afd0e50476e273bd0269a86/raw/7875f70561e1417721ec7f637fa6f160a27822a5/UsernameandPassword.png[Git Username and Password]
```